<launch>

	<arg name="use_sim_time" default="false" />
	<arg name="lidar_topic" default="/scan_front_multilayer/points_filtered" />
	<arg name="alpha" default="0.9" />

	<arg name="min_x" default="0.5" />
	<arg name="max_x" default="22.0" />
	<arg name="min_y" default="-2.0" />
	<arg name="max_y" default="1.5" />

	<arg name="hit_probability" default="0.9" />
	<arg name="miss_probability" default="0.4" />
	<arg name="max_range" default="5.0" />

	<node pkg="octomap_server" exec="octomap_server_node" name="octomap_server" namespace="canopy">
		<remap from="cloud_in" to="$(var lidar_topic)" />
		<param name="use_sim_time" value="$(var use_sim_time)" />
		<param name="resolution" value="0.2" />

		<param name="color.r" value="0.5" />
		<param name="color.g" value="1.0" />
		<param name="color.b" value="0.5" />
		<param name="color.a" value="$(var alpha)" />

		<param name="frame_id" value="hedge" />

		<param name="point_cloud_min_x" value="$(var min_x)" />
		<param name="point_cloud_max_x" value="$(var max_x)" />
		<param name="point_cloud_min_y" value="$(var min_y)" />
		<param name="point_cloud_max_y" value="$(var max_y)" />
		<param name="point_cloud_min_z" value="0.3" />

		<param name="sensor_model.hit" value="$(var hit_probability)" />
		<param name="sensor_model.miss" value="$(var miss_probability)" />
		<param name="sensor_model.max_range" value="$(var max_range)" />
	</node>

</launch>
